url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/algorithms_and_systems_AI_Evolutionary/Phrase.java

```diff
@@ -0,0 +1,243 @@
+// ##################################################################
+// ##################################################################
+// # Phrase: Based on a chromosome's genes it generates a musical 	#
+// # 							phrase								#
+// ##################################################################
+// ##################################################################
+package algorithms_and_systems_AI_Evolutionary;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import expert_system.Dynamic;
+import expert_system.DynamicShape;
+import expert_system.Key;
+import expert_system.Rhythmic_Block;
+import expert_system.Rhythmic_Value;
+import expert_system.Scale;
+
+
+public class Phrase {
+	
+	// ##########################################################
+	// #					      FIELDS						#
+	// ##########################################################
+	
+	public DynamicShape theShape;
+	
+	public List <Rhythmic_Block> rhythm;
+	
+	public int [] availableMidiNoteValues;
+	public int [] availableMidiVelocityValues;
+	
+	public Note [] musicalPhrase;
+	
+	// ##########################################################
+	// #					   CONSTRUCTORS						#
+	// ##########################################################
+
+	public Phrase(int min, int max, Scale s, Dynamic d, DynamicShape ds, ArrayList <Rhythmic_Block> arr) {
+		
+		this.theShape = ds;
+		this.availableMidiNoteValues = s.getScaleInRange(min, max);
+		this.availableMidiVelocityValues = d.getVelocityRange();
+		this.rhythm = arr;
+		this.generatePhrase();
+	}
+
+	// Returns musicalPhrase field
+	public Note[] getMusicalPhrase() {
+		return musicalPhrase;
+	}
+
+	// ##########################################################
+	// #						 METHODS	 					#
+	// ##########################################################
+	
+	// Generates Musical phrase
+	public void generatePhrase(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 31. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 78-84, lines 87-92 into separate methods.